shaders/src/getters.fs

```diff
@@ -41,6 +41,18 @@ vec3 getWorldViewVector() {
     return shading_view;
 }
 
+/** @public-api */
+vec3 getWorldViewVectorWithMagnitude() {
+    if (frameUniforms.clipFromViewMatrix[2].w != 0.0) {
```

Comment:
Nit: I would have written it as
```
    vec3 delta = shading_position - frameUniforms.cameraPosition;
    // Either perspective camera, or distance from plane (orthoEyePos, orthoViewDir)
    return (frameUniforms.clipFromViewMatrix[2].w != 0.0) ? delta : dot(shading_view, delta) * shading_view
```
due to my intense fear of shader compilers doing odd things unless given no alternatives. As in, there's no reason to branch here, so let's try to corner the compiler into generating a conditional move. Mind you, the above code still does not 100% guarantee that but it gives less room to breath.

Reply:
I kept the if/else because it is widely used in other Filament shaders, plus it is easier to comment the code with the separeted code blocks.